plaidml/edsl/edsl.h

```diff
@@ -657,10 +657,19 @@ class Tensor {
   ///
   /// Return the tensor's shape
   ///
-  LogicalShape shape() const {
+  LogicalShape compute_shape() const {
     return LogicalShape(ffi::call<plaidml_logical_shape*>(plaidml_expr_get_shape, as_ptr()));
   }
 
+  DType dtype() const {  //
```

Comment:
I'm excited about these `dtype` and `rank` functions!

Reply:
Ya, they basically make constructing resnet50 very fast, instead of always recomputing a shape we never ended up needing.